lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -128,16 +128,53 @@ public AuthOptions(String key) throws AblyException {
 		}
 
 		/**
-		 * Internal
+		 * Internal, when AuthOptions are provided, the values of each attribute are not merged,
+		 * but instead replace all corresponding values, even when null.
+		 *
+		 * should be used for:
+		 * - {@link Auth#requestToken(TokenParams, AuthOptions)} (Spec:RSA8e),
+		 * - {@link Auth#createTokenRequest(AuthOptions, TokenParams)} (Spec:RSA9h),
+		 * - {@link Auth#authorise(AuthOptions, TokenParams)} (Spec:RSA10j).
+		 *
+		 * <p>
+		 * Spec: RSA8e, RSA9h, RSA10j
+		 * </p>
+		 *
+		 * @param options if the object is null, the client library configured defaults are used.
+		 * @return superseded object
 		 */
-		public AuthOptions merge(AuthOptions defaults) {
-			if(authCallback == null) authCallback = defaults.authCallback;
-			if(authUrl == null) authUrl = defaults.authUrl;
-			if(key == null) key = defaults.key;
-			if(authHeaders == null) authHeaders = defaults.authHeaders;
-			if(authParams == null) authParams = defaults.authParams;
-			queryTime = queryTime & defaults.queryTime;
-			return this;
+		public AuthOptions replace(AuthOptions options) {
+			AuthOptions copy;
+			if (this instanceof ClientOptions) {
```

Comment:
Sorry, not sure I understand this logic, can you explain why this would be a `ClientOptions`?